internal/cli/atlas/accesslists/create.go

```diff
@@ -66,21 +68,24 @@ func (opts *CreateOpts) Run() error {
 	return opts.Print(r)
 }
 
-func (opts *CreateOpts) newProjectIPAccessList() []*atlas.ProjectIPAccessList {
-	entry := &atlas.ProjectIPAccessList{
-		GroupID:         opts.ConfigProjectID(),
-		Comment:         opts.comment,
-		DeleteAfterDate: opts.deleteAfter,
+func (opts *CreateOpts) newProjectIPAccessList() []*atlasv2.NetworkPermissionEntry {
+	entry := &atlasv2.NetworkPermissionEntry{
+		GroupId: pointer.Get(opts.ConfigProjectID()),
+		Comment: &opts.comment,
+	}
+	if opts.deleteAfter != "" {
+		deleteAfterDate, _ := time.Parse(time.RFC3339, opts.deleteAfter)
+		entry.DeleteAfterDate = &deleteAfterDate
```

Comment:
assuming this is because the new sdk uses a time type instead of a string type so, has anybody tried using https://pkg.go.dev/github.com/spf13/pflag#Duration for the flag and avoid this casting?

Reply:
created this story to follow up
https://jira.mongodb.org/browse/CLOUDP-169993